pkg/upstream/upstream.go

```diff
@@ -58,13 +61,20 @@ type Upstream struct {
 	PDClock     pdutil.Clock
 	GCManager   gc.Manager
 
+	// holder count of this upstream
+	hc int32
+	// record the time when Upstream.hc becomes zero.
+	idleTime time.Time
+	// mu should be lock when Upstream.hc or Upstream.idleTime need to be changed.
+	mu sync.Mutex
```

Comment:
Perhaps these three fields should be combined into single struct.
ref:https://github.com/pingcap/tiflow/pull/5264/files#diff-cfe8ccb173f8b93acc1b97cc5825fdb565d539634714fadffacdf18ea4bc8b26L145-R33